internal/expr/expr_test.go

```diff
@@ -449,6 +449,64 @@ func (s *ExprSuite) TestExprPkg(c *C) {
+	}, {
+		summary:        "insert specified columns to multiple types",
+		query:          "INSERT INTO person (id, street) VALUES ($Address.*, $M.street)",
+		expectedParsed: "[Bypass[INSERT INTO person ] ColumnInput[[id street] [Address.* M.street]]]",
+	}, {
+		summary:        "insert specified columns to multiple structs",
+		query:          "INSERT INTO person (name, street) VALUES ($Address.*, $Person.*)",
+		expectedParsed: "[Bypass[INSERT INTO person ] ColumnInput[[name street] [Address.* Person.*]]]",
+	}, {
+		summary:        "insert asterisk with comment",
+		query:          "INSERT INTO person (*) VALUES ($Person.address_id, /* rouge comment */$Address.street)",
+		expectedParsed: "[Bypass[INSERT INTO person ] AsteriskInput[[*] [Person.address_id Address.street]]]",
+	}, {
+		summary:        "insert asterisk (no space)",
+		query:          "INSERT INTO person(*) VALUES ($Person.*)ON CONFLICT DO NOTHING",
+		expectedParsed: "[Bypass[INSERT INTO person] AsteriskInput[[*] [Person.*]] Bypass[ON CONFLICT DO NOTHING]]",
+	}, {
+		summary:        "insert asterisk (weird spacing)",
+		query:          "INSERT INTO person ( * )VALUES( $Person.* )ON CONFLICT DO NOTHING",
+		expectedParsed: "[Bypass[INSERT INTO person ] AsteriskInput[[*] [Person.*]] Bypass[ON CONFLICT DO NOTHING]]",
+	}, {
+		summary:        "insert with returning clause",
+		query:          "INSERT INTO address(*) VALUES($Address.*) RETURNING (&Address.*)",
+		expectedParsed: "[Bypass[INSERT INTO address] AsteriskInput[[*] [Address.*]] Bypass[ RETURNING (] Output[[] [Address.*]] Bypass[)]]",
+	}, {
+		summary:        "insert rename columns with standalone inputs",
+		query:          `INSERT INTO person (id, random_string, random_thing, street) VALUES ($Person.address_id, "random string", rand(), $Address.street)`,
+		expectedParsed: `[Bypass[INSERT INTO person (id, random_string, random_thing, street) VALUES (] Input[Person.address_id] Bypass[, "random string", rand(), ] Input[Address.street] Bypass[)]]`,
+	}, {
+		summary:        "insert single value",
+		query:          "INSERT INTO person (name) VALUES ($Person.name)",
+		expectedParsed: "[Bypass[INSERT INTO person (name) VALUES (] Input[Person.name] Bypass[)]]",
+	}, {
+		summary:        "insert with standalone input expressions",
+		query:          `INSERT INTO person VALUES ($Person.name, "random string", $Person.id)`,
+		expectedParsed: `[Bypass[INSERT INTO person VALUES (] Input[Person.name] Bypass[, "random string", ] Input[Person.id] Bypass[)]]`,
+	}}
+	for i, t := range tests {
+		parser := expr.NewParser()
+		expr, err := parser.Parse(t.query)
+		c.Assert(err, IsNil, Commentf("test %d failed (Parse):\nsummary: %s\nquery: %s\nexpected: %s\nerr: %s\n", i, t.summary, t.query, t.expectedParsed, err))
+		c.Assert(expr.String(), Equals, t.expectedParsed, Commentf("test %d failed (Parse):\nsummary: %s\nquery: %s\nexpected: %s\nactual:   %s\n", i, t.summary, t.query, t.expectedParsed, expr.String()))
```

Comment:
[Not even Nit]: Lately I have seen tests in Chisel using `c.Log(test.summary)` and keeping the asserts bare, that seems to work because the logs are only printed when the test fails. Seems like an interesting pattern.

Reply:
That's seems like a better pattern, we should change it to do that at some point, though I won't do it in this PR